### PR TITLE
[REF] refactor test plotting

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Closes none

Changes proposed in this pull request:

- refactor several tests for plotting to make them smaller
- use pyplot fixture to clean up figures on teardown

### CODE REVIEW

File: nilearn/plotting/tests/test_img_plotting/test_plot_carpet.py [hunk 1]
```diff
@@ -46,12 +51,10 @@ def test_plot_carpet(img_4d_mni, img_3d_ones_mni, img_4d_long_mni):
         img_4d_long_mni.shape[-1] / 2
     )
     assert plotted_array.size == n_items
-    plt.close(display)
 
 
-def test_plot_carpet_with_atlas(img_4d_mni, img_atlas):
+def test_plot_carpet_with_atlas(pyplot, img_4d_mni, img_atlas):
```
Comment (bthirion) [inline nilearn/plotting/tests/test_img_plotting/test_plot_carpet.py:56]: Sorry, what is the point ?

Comment (Remi-Gau) [reply nilearn/plotting/tests/test_img_plotting/test_plot_carpet.py:56]: This uses the same fixture as in sklearn: 

https://scikit-learn.org/dev/developers/contributing.html#testing-and-improving-test-coverage

image

The fixture is in nilearn/plotting/tests/conftest.py

- checks if matplotlib is there, skips test if not
- setup: close any open figure before the test is run
- teardown: closes any figure open by the test

hence why I am removing all the plt.close('all') calls: the fixture does it for us even if the test fails: this makes sure tests are more independent and do not pollute each other

**Review by hndgzkn — APPROVED** [review]
It seems OK to me.

There are a few points, may be to consider for follow-up PRs:

- It might be an option to move it to `nilearn/conftest.py` file as there might be tests in other packages which might reuse `pyplot`.
- In that case will we still need this line? https://github.com/nilearn/nilearn/blob/87ab7395669320e8aeadc512f60093ba7df88aee/nilearn/conftest.py#L90 
- and this line? https://github.com/nilearn/nilearn/blob/87ab7395669320e8aeadc512f60093ba7df88aee/nilearn/conftest.py#L33

File: nilearn/plotting/tests/test_img_plotting/test_plot_carpet.py (hunk 1 above)
Comment (Remi-Gau) [reply nilearn/plotting/tests/test_img_plotting/test_plot_carpet.py:56]: probably worth adding something similar in our own doc

**Comment by Remi-Gau** [comment]
> actually the close_all fixture is not used so we can probably just remove it

nope my bad: it is ALWAYS used for EVERY test even if plotting is not involved, so maybe better to have a fixture we explicitly call for tests that need clean up.

**Review by bthirion — COMMENTED** [review]
Is this ready for merge ? LGTM.

**Comment by Remi-Gau** [comment]
Yup ready for merge.

**Pull request merged**